File: demos/ios/Demos/Demos/views/BaseBarChart2.m
//
//  BaseInterval2UIView.m
//  Demos
//
//  Created by XiaLuo on 2020/11/5.
//  Copyright © 2020 com.alipay.xgraph. All rights reserved.
//

#import "BaseBarChart2.h"

@implementation BaseBarChart2

- (NSString *)jsonData {
    return @"[{\"genre\":\"Sports\",\"sold\":275},{\"genre\":\"Strategy\",\"sold\":115},{\"genre\":\"Action\",\"sold\":120},{"
           @"\"genre\":\"Shooter\",\"sold\":350},{\"genre\":\"Other\",\"sold\":150}]";
}

- (void)chartRender {
    self.chart.canvas(self.canvasView);
    self.chart.padding(10, 20, 10, 0.f);
    self.chart.source([F2Utils toJsonArray:[self jsonData]]);
    self.chart.interval().position(@"genre*sold").color(@"genre", @[]);
    self.chart.tooltip(@{});
    self.chart.render();
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/BaseBarChart2.swift
//
//  BaseBarChart2.swift
//  F2Native
//
//  Created by Westin on 2024-06-09.
//

import F2

class BaseBarChart2: BaseLineChart {
    
    override func chartRender() {
        let jsonPath = Bundle.main.path(forResource: "Res/mockData_baseInterval2", ofType: "json")
        guard let jsonString = try? String.init(contentsOfFile: jsonPath!) else {
            return
        }
        let jsonData = F2Utils.toJsonArray(jsonString)
        self.chart!.canvas()(self.canvasView!)
        self.chart!.padding()(10, 20, 10, 0)
        self.chart!.source()(jsonData)
        self.chart!.interval()().position()("genre*sold").color()("genre", [])
        self.chart!.scale()("sold", ["min":0])
        self.chart!.scale()("genre", ["range": [0.1, 0.9]])
        self.chart!.animate()(true)
        self.chart!.render()();
    }
}
